xmr-btc/tests/harness/storage.rs

```diff
@@ -0,0 +1,158 @@
+use anyhow::{anyhow, Context, Result};
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use std::path::Path;
+
+pub struct Database<T>
+where
+    T: Serialize + DeserializeOwned,
+{
+    db: sled::Db,
+    _marker: std::marker::PhantomData<T>,
+}
+
+impl<T> Database<T>
+where
+    T: Serialize + DeserializeOwned,
+{
+    const LAST_STATE_KEY: &'static str = "latest_state";
+
+    pub fn open(path: &Path) -> Result<Self> {
+        let db =
+            sled::open(path).with_context(|| format!("Could not open the DB at {:?}", path))?;
+
+        Ok(Database {
+            db,
+            _marker: Default::default(),
+        })
+    }
+
+    pub async fn insert_latest_state(&self, state: &T) -> Result<()> {
+        let key = serialize(&Self::LAST_STATE_KEY)?;
```

Comment:
As mentioned previously, use `conquer_once::Lazy` to store this value directly serialised in a static variable so you can't forget to serialise it when using it.

Reply:
added todo